app/services/create_subscriber_list_service.rb

```diff
@@ -0,0 +1,54 @@
+class CreateSubscriberListService < ApplicationService
+  def initialize(params:, user:)
+    @params = params
+    @user = user
+  end
+
+  def call
+    subscriber_list = FindExactQuery.new(**find_exact_query_params).exact_match
+    return SubscriberList.create!(subscriber_list_params) unless subscriber_list
+
+    subscriber_list.update!(subscriber_list_params.slice(:title, :url).compact)
+    subscriber_list
+  end
+
+private
+
+  attr_reader :params, :user
+
+  def subscriber_list_params
+    find_exact_query_params.merge(
+      title: params[:title],
+      slug: slug,
+      url: params[:url],
+      signon_user_uid: user.uid,
+    )
+  end
+
+  def convert_legacy_params(link_or_tags)
+    link_or_tags.transform_values do |link_or_tag|
+      link_or_tag.is_a?(Hash) ? link_or_tag : { any: link_or_tag }
+    end
+  end
+
+  def find_exact_query_params
```

Comment:
do you think we need to do anything (perhaps in future) to address this duplication? I'm not sure these keys change very often, and I don't mind a bit, so long as it doesn't bite me :-)

Reply:
(I assume you're referring to the duplication in the controller.) Yes, once we switch [the adapter](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/email_alert_api.rb#L12) to just use the "create" endpoint, the "index" action, and its associated params/methods, can be deleted i.e. the duplication should only be temporary.

Reply:
Perfect 👍